demos/jobs/jobs_demo_mosquitto/jobs_demo_mosquitto.c

```diff
@@ -0,0 +1,1264 @@
+
+/**
+ * @brief publish a request to the Jobs service to start a job
+ *
+ * @param[in] h runtime state handle
+ *
+ * @return true if libmosquitto accepted the publish message;
+ * false otherwise
+ *
+ * @note This does not call mosquitto_loop(); it expects main() to do so.
+ */
+static bool sendStartNext( handle_t * h );
+
+/**
+ * @brief reports status of the download process
+ *
+ * @param[in] h runtime state handle
+ *
+ * @return the result of sendUpdate()
+ */
+static bool update( handle_t * h );
+
+/**
+ * @brief launch a download process
+ *
+ * @param[in] h runtime state handle
+ *
+ * @return true if fork() succeeded;
+ * false otherwise
+ */
+static bool download( handle_t * h );
+
+/**
+ * @brief kill a download process
+ *
+ * @param[in] h runtime state handle
+ */
+static void cancelDownload( handle_t * h );
+
+/**
+ * @brief libmosquitto callback for log messages
+ *
+ * @param[in] m unused
+ * @param[in] p unused
+ * @param[in] level unused
+ * @param[in] log the message to print
+ */
+static void on_log( struct mosquitto * m,
+                    void * p,
+                    int level,
+                    const char * log );
+
+/**
+ * @brief generic signal handler
+ *
+ * @param[in] signal the caught signal value
+ */
+static void catch( int signal );
+
+/**
+ * @brief setup signal handling and libmosquitto
+ *
+ * @param[in] h runtime state handle
+ *
+ * @return false if a step failed;
+ * true otherwise
+ */
+static bool setup( handle_t * h );
+
+/**
+ * @brief disconnect and clean up
+ *
+ * @param[in] x unused
+ * @param[in] p runtime state handle
+ */
+static void teardown( int x,
+                      void * p );
+
+/**
+ * @brief log an informational message
+ */
+#define info    warnx
+
+/**
+ * @brief format a JSON status message
+ *
+ * @param[in] x one of "IN_PROGRESS", "SUCCEEDED", or "FAILED"
+ */
+#define makeReport_( x )    "{\"status\":\"" x "\"}"
+
+/*-----------------------------------------------------------*/
+
+void initHandle( handle_t * p )
+{
+    assert( p != NULL );
+
+    handle_t h = { 0 };
+    h.port = DEFAULT_MQTT_PORT;
+    h.capath = DEFAULT_CA_DIRECTORY;
+    /* initialize to -1, set by on_connect() to 0 or greater */
+    h.connectError = -1;
+    /* initialize to -1, set by on_subscribe() to 0 or greater */
+    h.subscribeQOS = -1;
+
+    *p = h;
+}
+
+/*-----------------------------------------------------------*/
+
+static bool requiredArgs( handle_t * h )
+{
+    bool ret = true;
+    struct stat s;
+
+    assert( h != NULL );
+
+#define checkString( x )                               \
+    if( ( h->x == NULL ) || ( h->x[ 0 ] == '\0' ) )    \
+    {                                                  \
+        ret = false;                                   \
+        warnx( "%s argument must not be empty", # x ); \
+    }
+
+    checkString( name );
+    checkString( host );
+    checkString( cert );
+    checkString( key );
```

Comment:
It's nice that you have added CLI arguments for this demo. In fact, I want to add this option for all the other demos too. We also allow CLI parameters to be passed through CMake so that these parameters can be defined for ALL demos. Would it then be possible to set the value of name, host, cert, key if they are passed through CMake? We use the CMake function `target_compile_definitions` for this.

Reply:
Sure, I added AWS_IOT_ENDPOINT, ROOT_CA_CERT_PATH, CLIENT_CERT_PATH, and CLIENT_PRIVATE_KEY_PATH.

Reply:
It seems that these weren't added to the `CMakeLists.txt`

Reply:
fixed